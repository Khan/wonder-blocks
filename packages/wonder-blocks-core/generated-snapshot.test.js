// This file is auto-generated by gen-snapshot-tests.js
// Do not edit this file.  To make changes to these snapshot tests:
//   1. edit the markdown documentation files in the package,
//        packages/wonder-blocks-core
//   2. Run `yarn run gen-snapshot-tests`.
import React from "react";
import renderer from "react-test-renderer";

// Mock react-dom as jest doesn't like findDOMNode.
jest.mock("react-dom");
import ClickableBehavior from "./components/clickable-behavior.js";
import Clickable from "./components/clickable.js";
import IDProvider from "./components/id-provider.js";
import Text from "./components/text.js";
import UniqueIDProvider from "./components/unique-id-provider.js";
import View from "./components/view.js";
import WithSSRPlaceholder from "./components/with-ssr-placeholder.js";

describe("wonder-blocks-core", () => {
    it("example 1", () => {
        const {
            LabelLarge,
            LabelMedium,
        } = require("@khanacademy/wonder-blocks-typography");
        const Icon = require("@khanacademy/wonder-blocks-icon").default;
        const {icons} = require("@khanacademy/wonder-blocks-icon");
        const {Spring, Strut} = require("@khanacademy/wonder-blocks-layout");
        const {StyleSheet} = require("aphrodite");

        const styles = StyleSheet.create({
            container: {
                display: "flex",
                flexDirection: "row",
                padding: 16,
                border: "solid 1px black",
                borderRadius: 4,
                width: "100%",
            },
            hovered: {
                backgroundColor: "lightgreen",
                cursor: "pointer",
            },
            pressed: {
                backgroundColor: "green",
                color: "white",
            },
            focused: {
                "::before": {
                    content: "''",
                    position: "absolute",
                    left: -8,
                    top: -8,
                    right: -8,
                    bottom: -8,
                    borderRadius: 10,
                    border: "dotted 4px black",
                },
            },
        });

        const example = (
            <View>
                <Clickable>
                    {(state) => {
                        console.log(state);
                        return (
                            <View
                                style={[
                                    styles.container,
                                    state.hovered && styles.hovered,
                                    state.pressed && styles.pressed,
                                    state.focused && styles.focused,
                                ]}
                            >
                                <View style={{alignItems: "flex-start"}}>
                                    <LabelLarge>Hello, world</LabelLarge>
                                    <Strut size={8} />
                                    <LabelMedium>Lorem Ipsum Dolor</LabelMedium>
                                </View>
                                <Spring />
                                <Icon size="large" icon={icons.search} />
                            </View>
                        );
                    }}
                </Clickable>
            </View>
        );
        const tree = renderer.create(example).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("example 2", () => {
        const example = (
            <View>
                <IDProvider scope="field">
                    {(uniqueId) => (
                        <label htmlFor={uniqueId}>
                            Label with ID {uniqueId}:
                            <input type="text" id={uniqueId} />
                        </label>
                    )}
                </IDProvider>
            </View>
        );
        const tree = renderer.create(example).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("example 3", () => {
        const example = (
            <View>
                <IDProvider scope="field" id="some-user-id">
                    {(uniqueId) => (
                        <label htmlFor={uniqueId}>
                            Label with ID {uniqueId}:
                            <input type="text" id={uniqueId} />
                        </label>
                    )}
                </IDProvider>
            </View>
        );
        const tree = renderer.create(example).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("example 4", () => {
        const {StyleSheet} = require("aphrodite");

        const styles = StyleSheet.create({
            container: {
                padding: 32,
                backgroundColor: "plum",
            },
            text: {
                fontFamily: "sans-serif",
                fontSize: 24,
            },
        });

        const example = (
            <View style={styles.container}>
                <Text style={styles.text}>Hello, world!</Text>
                <View
                    style={[
                        styles.container,
                        {backgroundColor: "goldenrod", padding: 4},
                    ]}
                >
                    Even with an array and inline styles!
                </View>
            </View>
        );
        const tree = renderer.create(example).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("example 5", () => {
        const example = (
            <View>
                <View onClick={() => alert("Clicked!")}>Click me!</View>

                <Text aria-hidden="true">
                    This text is hidden from screen readers.
                </Text>
            </View>
        );
        const tree = renderer.create(example).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("example 6", () => {
        const example = (
            <View>
                <View testId="foo">Foo</View>
                <Text testId="bar">Bar</Text>
            </View>
        );
        const tree = renderer.create(example).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("example 7", () => {
        const {
            Body,
            HeadingSmall,
        } = require("@khanacademy/wonder-blocks-typography");
        const {Spring, Strut} = require("@khanacademy/wonder-blocks-layout");
        const Button = require("@khanacademy/wonder-blocks-button").default;

        let providerRef = null;

        const renders = [];
        const provider = (
            <UniqueIDProvider
                mockOnFirstRender={false}
                ref={(ref) => (providerRef = ref)}
            >
                {(ids) => {
                    renders.push(ids.get("my-unique-id"));
                    return (
                        <View>
                            {renders.map((id, i) => (
                                <Body key={i}>
                                    Render {i}: {id}
                                </Body>
                            ))}
                        </View>
                    );
                }}
            </UniqueIDProvider>
        );

        const onClick = () => {
            if (providerRef) {
                providerRef.forceUpdate();
            }
        };

        const example = (
            <View>
                <View style={{flexDirection: "row"}}>
                    <Button onClick={onClick}>Click Me to Rerender</Button>
                    <Spring />
                </View>
                <Strut size={16} />
                <HeadingSmall>The UniqueIDProvider:</HeadingSmall>
                {provider}
            </View>
        );
        const tree = renderer.create(example).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("example 8", () => {
        const {
            Body,
            BodyMonospace,
            HeadingSmall,
        } = require("@khanacademy/wonder-blocks-typography");
        const {Spring, Strut} = require("@khanacademy/wonder-blocks-layout");

        let firstId = null;
        let secondId = null;

        const children = (idf) => {
            const id1 = idf.get("an-id");
            const id2 = idf.get("something");
            firstId = firstId || id1;
            secondId = secondId || id2;
            return (
                <View>
                    <HeadingSmall>The initial render:</HeadingSmall>
                    <View>
                        <BodyMonospace>get("an-id"): {firstId}</BodyMonospace>
                        <BodyMonospace>
                            get("something"): {secondId}
                        </BodyMonospace>
                    </View>
                    <Strut size={16} />
                    <HeadingSmall>Subsequent requests:</HeadingSmall>
                    <View>
                        <BodyMonospace>get("an-id"): {id1}</BodyMonospace>
                        <BodyMonospace>get("something"): {id2}</BodyMonospace>
                    </View>
                </View>
            );
        };

        const example = (
            <UniqueIDProvider mockOnFirstRender={true}>
                {children}
            </UniqueIDProvider>
        );
        const tree = renderer.create(example).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("example 9", () => {
        const {
            Body,
            HeadingSmall,
            BodyMonospace,
        } = require("@khanacademy/wonder-blocks-typography");
        const {Spring, Strut} = require("@khanacademy/wonder-blocks-layout");

        const children = ({get}) => (
            <View>
                <Body>
                    The id returned for "my-identifier": {get("my-identifier")}
                </Body>
            </View>
        );

        const example = (
            <View>
                <HeadingSmall>First Provider with scope: first</HeadingSmall>
                <UniqueIDProvider mockOnFirstRender={false} scope="first">
                    {children}
                </UniqueIDProvider>
                <HeadingSmall>Second Provider with scope: second</HeadingSmall>
                <UniqueIDProvider mockOnFirstRender={false} scope="second">
                    {children}
                </UniqueIDProvider>
            </View>
        );
        const tree = renderer.create(example).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("example 10", () => {
        const {
            BodyMonospace,
        } = require("@khanacademy/wonder-blocks-typography");
        const {Strut} = require("@khanacademy/wonder-blocks-layout");

        // TODO(somewhatabstract): Update this to be nice once we can get BodyMonospace
        // to allow us to properly preserve whitespace or have an alternative. Or remove
        // this entirely when our styleguide renders our interface definitions.
        const example = (
            <View>
                <BodyMonospace>
                    interface IIdentifierFactory &#123;
                </BodyMonospace>
                <View style={{flexDirection: "row"}}>
                    <Strut size={"2em"} />
                    <BodyMonospace>get(id: string): string;</BodyMonospace>
                </View>
                <BodyMonospace>&#125;</BodyMonospace>
            </View>
        );
        const tree = renderer.create(example).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("example 11", () => {
        const {StyleSheet} = require("aphrodite");

        const styles = StyleSheet.create({
            container: {
                padding: 32,
                backgroundColor: "plum",
            },
            text: {
                fontFamily: "sans-serif",
                fontSize: 24,
            },
        });

        const example = (
            <View style={styles.container}>
                <Text style={styles.text}>Hello, world!</Text>
                <View
                    style={[
                        styles.container,
                        {backgroundColor: "goldenrod", padding: 4},
                    ]}
                >
                    Even with an array and inline styles!
                </View>
            </View>
        );
        const tree = renderer.create(example).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("example 12", () => {
        const example = (
            <View>
                <View onClick={() => alert("Clicked!")}>Click me!</View>

                <Text aria-hidden="true">
                    This text is hidden from screen readers.
                </Text>
            </View>
        );
        const tree = renderer.create(example).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("example 13", () => {
        const example = (
            <WithSSRPlaceholder
                placeholder={() => (
                    <View>
                        This gets rendered on server, and also on the client for
                        the very first render (the "rehydration" render)
                    </View>
                )}
            >
                {() => (
                    <View>
                        This is rendered only by the client, for all renders
                        after the rehydration render.
                    </View>
                )}
            </WithSSRPlaceholder>
        );
        const tree = renderer.create(example).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("example 14", () => {
        const example = (
            <WithSSRPlaceholder placeholder={null}>
                {() => (
                    <View>
                        This is rendered only by the client, while nothing was
                        rendered on the server.
                    </View>
                )}
            </WithSSRPlaceholder>
        );
        const tree = renderer.create(example).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("example 15", () => {
        const {
            Body,
            BodyMonospace,
        } = require("@khanacademy/wonder-blocks-typography");

        const trackingArray = [];
        const resultsId = "nossr-example-2-results";
        const newLi = (text) => {
            const li = document.createElement("li");
            li.appendChild(document.createTextNode(text));
            return li;
        };

        const addTrackedRender = (text) => {
            const el = document.getElementById(resultsId);
            if (el) {
                for (let i = 0; i < trackingArray.length; i++) {
                    el.append(newLi(trackingArray[i]));
                }
                trackingArray.length = 0;
                el.append(newLi(text));
            } else {
                // We may not have rendered the results element yet, so if we haven't
                // use an array to keep track of the things until we have.
                trackingArray.push(text);
            }
        };

        const trackAndRender = (text) => {
            addTrackedRender(text);
            return text;
        };

        const example = (
            <View>
                <Body>
                    The list below should have three render entries; root
                    placeholder, root children render, and child children
                    render. If there are two child renders that means that the
                    second forced render is still occurring for nested
                    WithSSRPlaceholder components, which would be a bug.
                </Body>
                <ul id={resultsId} />
                <Body>
                    And below this is the actual WithSSRPlaceholder nesting,
                    which should just show the child render.
                </Body>
                <WithSSRPlaceholder
                    placeholder={() => (
                        <View>{trackAndRender("Root: placeholder")}</View>
                    )}
                >
                    {() => {
                        addTrackedRender("Root: render");
                        return (
                            <WithSSRPlaceholder
                                placeholder={() => (
                                    <View>
                                        {trackAndRender(
                                            "Child: placeholder (should never see me)",
                                        )}
                                    </View>
                                )}
                            >
                                {() => (
                                    <View>
                                        {trackAndRender("Child: render")}
                                    </View>
                                )}
                            </WithSSRPlaceholder>
                        );
                    }}
                </WithSSRPlaceholder>
            </View>
        );
        const tree = renderer.create(example).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("example 16", () => {
        const {
            Body,
            BodyMonospace,
        } = require("@khanacademy/wonder-blocks-typography");

        const trackingArray = [];
        const resultsId = "nossr-example-3-results";
        const newLi = (text) => {
            const li = document.createElement("li");
            li.appendChild(document.createTextNode(text));
            return li;
        };

        const addTrackedRender = (text) => {
            const el = document.getElementById(resultsId);
            if (el) {
                for (let i = 0; i < trackingArray.length; i++) {
                    el.append(newLi(trackingArray[i]));
                }
                trackingArray.length = 0;
                el.append(newLi(text));
            } else {
                // We may not have rendered the results element yet, so if we haven't
                // use an array to keep track of the things until we have.
                trackingArray.push(text);
            }
        };

        const trackAndRender = (text) => {
            addTrackedRender(text);
            return text;
        };

        const example = (
            <View>
                <Body>
                    The list below should have six render entries; 2 x root
                    placeholder, 2 x root children render, and 2 x child
                    children render.
                </Body>
                <ul id={resultsId} />
                <Body>
                    And below this are the WithSSRPlaceholder component trees,
                    which should just show their child renders.
                </Body>
                <WithSSRPlaceholder
                    placeholder={() => (
                        <View>{trackAndRender("Root 1: placeholder")}</View>
                    )}
                >
                    {() => {
                        addTrackedRender("Root 1: render");
                        return (
                            <WithSSRPlaceholder
                                placeholder={() => (
                                    <View>
                                        {trackAndRender(
                                            "Child 1: placeholder (should never see me)",
                                        )}
                                    </View>
                                )}
                            >
                                {() => (
                                    <View>
                                        {trackAndRender("Child 1: render")}
                                    </View>
                                )}
                            </WithSSRPlaceholder>
                        );
                    }}
                </WithSSRPlaceholder>
                <WithSSRPlaceholder
                    placeholder={() => (
                        <View>{trackAndRender("Root 2: placeholder")}</View>
                    )}
                >
                    {() => {
                        addTrackedRender("Root 2: render");
                        return (
                            <WithSSRPlaceholder
                                placeholder={() => (
                                    <View>
                                        {trackAndRender(
                                            "Child 2: placeholder (should never see me)",
                                        )}
                                    </View>
                                )}
                            >
                                {() => (
                                    <View>
                                        {trackAndRender("Child 2: render")}
                                    </View>
                                )}
                            </WithSSRPlaceholder>
                        );
                    }}
                </WithSSRPlaceholder>
            </View>
        );
        const tree = renderer.create(example).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
