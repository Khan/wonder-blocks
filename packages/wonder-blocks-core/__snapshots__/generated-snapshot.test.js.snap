// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wonder-blocks-core example 1 1`] = `
<div
  className=""
  style={
    Object {
      "alignItems": "stretch",
      "borderStyle": "solid",
      "borderWidth": 0,
      "boxSizing": "border-box",
      "display": "flex",
      "flexDirection": "column",
      "margin": 0,
      "minHeight": 0,
      "minWidth": 0,
      "padding": 0,
      "position": "relative",
      "zIndex": 0,
    }
  }
>
  This is rendered only by the client for all but the first render.
</div>
`;

exports[`wonder-blocks-core example 2 1`] = `
<div
  className=""
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": "plum",
      "borderStyle": "solid",
      "borderWidth": 0,
      "boxSizing": "border-box",
      "display": "flex",
      "flexDirection": "column",
      "margin": 0,
      "minHeight": 0,
      "minWidth": 0,
      "padding": 32,
      "position": "relative",
      "zIndex": 0,
    }
  }
>
  <span
    className=""
    style={
      Object {
        "MozOsxFontSmoothing": "grayscale",
        "WebkitFontSmoothing": "antialiased",
        "fontFamily": "sans-serif",
        "fontSize": 24,
      }
    }
  >
    Hello, world!
  </span>
  <div
    className=""
    style={
      Object {
        "alignItems": "stretch",
        "backgroundColor": "goldenrod",
        "borderStyle": "solid",
        "borderWidth": 0,
        "boxSizing": "border-box",
        "display": "flex",
        "flexDirection": "column",
        "margin": 0,
        "minHeight": 0,
        "minWidth": 0,
        "padding": 4,
        "position": "relative",
        "zIndex": 0,
      }
    }
  >
    Even with an array and inline styles!
  </div>
</div>
`;

exports[`wonder-blocks-core example 3 1`] = `
<div
  className=""
  style={
    Object {
      "alignItems": "stretch",
      "borderStyle": "solid",
      "borderWidth": 0,
      "boxSizing": "border-box",
      "display": "flex",
      "flexDirection": "column",
      "margin": 0,
      "minHeight": 0,
      "minWidth": 0,
      "padding": 0,
      "position": "relative",
      "zIndex": 0,
    }
  }
>
  <div
    className=""
    onClick={[Function]}
    style={
      Object {
        "alignItems": "stretch",
        "borderStyle": "solid",
        "borderWidth": 0,
        "boxSizing": "border-box",
        "display": "flex",
        "flexDirection": "column",
        "margin": 0,
        "minHeight": 0,
        "minWidth": 0,
        "padding": 0,
        "position": "relative",
        "zIndex": 0,
      }
    }
  >
    Click me!
  </div>
  <span
    aria-hidden="true"
    className=""
    style={
      Object {
        "MozOsxFontSmoothing": "grayscale",
        "WebkitFontSmoothing": "antialiased",
      }
    }
  >
    This text is hidden from screen readers.
  </span>
</div>
`;

exports[`wonder-blocks-core example 4 1`] = `
<div
  className=""
  style={
    Object {
      "alignItems": "stretch",
      "borderStyle": "solid",
      "borderWidth": 0,
      "boxSizing": "border-box",
      "display": "flex",
      "flexDirection": "column",
      "margin": 0,
      "minHeight": 0,
      "minWidth": 0,
      "padding": 0,
      "position": "relative",
      "zIndex": 0,
    }
  }
>
  <div
    className=""
    style={
      Object {
        "alignItems": "stretch",
        "borderStyle": "solid",
        "borderWidth": 0,
        "boxSizing": "border-box",
        "display": "flex",
        "flexDirection": "row",
        "margin": 0,
        "minHeight": 0,
        "minWidth": 0,
        "padding": 0,
        "position": "relative",
        "zIndex": 0,
      }
    }
  >
    <button
      className=""
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "::MozFocusInner": Object {
            "border": 0,
          },
          "alignItems": "center",
          "background": "#1865f2",
          "border": "none",
          "borderRadius": 4,
          "boxSizing": "border-box",
          "color": "#ffffff",
          "cursor": "pointer",
          "display": "inline-flex",
          "height": 40,
          "justifyContent": "center",
          "margin": 0,
          "outline": "none",
          "paddingBottom": 0,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 0,
          "position": "relative",
          "textDecoration": "none",
        }
      }
      tabIndex={0}
    >
      <span
        className=""
        style={
          Object {
            "MozOsxFontSmoothing": "grayscale",
            "WebkitFontSmoothing": "antialiased",
            "display": "block",
            "fontFamily": "Lato, sans-serif",
            "fontSize": 16,
            "fontWeight": "bold",
            "lineHeight": "20px",
            "pointerEvents": "none",
          }
        }
      >
        Click Me to Rerender
      </span>
    </button>
    <div
      aria-hidden="true"
      className=""
      style={
        Object {
          "alignItems": "stretch",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexDirection": "column",
          "flexGrow": 1,
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "zIndex": 0,
        }
      }
    />
  </div>
  <div
    aria-hidden="true"
    className=""
    style={
      Object {
        "MsFlexBasis": 16,
        "MsFlexPreferredSize": 16,
        "WebkitFlexBasis": 16,
        "alignItems": "stretch",
        "borderStyle": "solid",
        "borderWidth": 0,
        "boxSizing": "border-box",
        "display": "flex",
        "flexBasis": 16,
        "flexDirection": "column",
        "flexShrink": 0,
        "margin": 0,
        "minHeight": 0,
        "minWidth": 0,
        "padding": 0,
        "position": "relative",
        "zIndex": 0,
      }
    }
  />
  <h4
    className=""
    style={
      Object {
        "MozOsxFontSmoothing": "grayscale",
        "WebkitFontSmoothing": "antialiased",
        "display": "block",
        "fontFamily": "Lato, sans-serif",
        "fontSize": 20,
        "fontWeight": 700,
        "lineHeight": "24px",
        "marginBottom": 0,
        "marginTop": 0,
      }
    }
  >
    The UniqueIDProvider:
  </h4>
  <div
    className=""
    style={
      Object {
        "alignItems": "stretch",
        "borderStyle": "solid",
        "borderWidth": 0,
        "boxSizing": "border-box",
        "display": "flex",
        "flexDirection": "column",
        "margin": 0,
        "minHeight": 0,
        "minWidth": 0,
        "padding": 0,
        "position": "relative",
        "zIndex": 0,
      }
    }
  >
    <span
      className=""
      style={
        Object {
          "MozOsxFontSmoothing": "grayscale",
          "WebkitFontSmoothing": "antialiased",
          "display": "block",
          "fontFamily": "Lato, sans-serif",
          "fontSize": 16,
          "fontWeight": 400,
          "lineHeight": "22px",
        }
      }
    >
      Render 
      0
      : 
      uid--0-my-unique-id
    </span>
  </div>
</div>
`;

exports[`wonder-blocks-core example 5 1`] = `
<div
  className=""
  style={
    Object {
      "alignItems": "stretch",
      "borderStyle": "solid",
      "borderWidth": 0,
      "boxSizing": "border-box",
      "display": "flex",
      "flexDirection": "column",
      "margin": 0,
      "minHeight": 0,
      "minWidth": 0,
      "padding": 0,
      "position": "relative",
      "zIndex": 0,
    }
  }
>
  <h4
    className=""
    style={
      Object {
        "MozOsxFontSmoothing": "grayscale",
        "WebkitFontSmoothing": "antialiased",
        "display": "block",
        "fontFamily": "Lato, sans-serif",
        "fontSize": 20,
        "fontWeight": 700,
        "lineHeight": "24px",
        "marginBottom": 0,
        "marginTop": 0,
      }
    }
  >
    The initial render:
  </h4>
  <div
    className=""
    style={
      Object {
        "alignItems": "stretch",
        "borderStyle": "solid",
        "borderWidth": 0,
        "boxSizing": "border-box",
        "display": "flex",
        "flexDirection": "column",
        "margin": 0,
        "minHeight": 0,
        "minWidth": 0,
        "padding": 0,
        "position": "relative",
        "zIndex": 0,
      }
    }
  >
    <span
      className=""
      style={
        Object {
          "MozOsxFontSmoothing": "grayscale",
          "WebkitFontSmoothing": "antialiased",
          "display": "block",
          "fontFamily": "Inconsolata, monospace",
          "fontSize": 17,
          "fontWeight": 400,
          "lineHeight": "22px",
        }
      }
    >
      get("an-id"): 
      an-id
    </span>
    <span
      className=""
      style={
        Object {
          "MozOsxFontSmoothing": "grayscale",
          "WebkitFontSmoothing": "antialiased",
          "display": "block",
          "fontFamily": "Inconsolata, monospace",
          "fontSize": 17,
          "fontWeight": 400,
          "lineHeight": "22px",
        }
      }
    >
      get("something"): 
      something
    </span>
  </div>
  <div
    aria-hidden="true"
    className=""
    style={
      Object {
        "MsFlexBasis": 16,
        "MsFlexPreferredSize": 16,
        "WebkitFlexBasis": 16,
        "alignItems": "stretch",
        "borderStyle": "solid",
        "borderWidth": 0,
        "boxSizing": "border-box",
        "display": "flex",
        "flexBasis": 16,
        "flexDirection": "column",
        "flexShrink": 0,
        "margin": 0,
        "minHeight": 0,
        "minWidth": 0,
        "padding": 0,
        "position": "relative",
        "zIndex": 0,
      }
    }
  />
  <h4
    className=""
    style={
      Object {
        "MozOsxFontSmoothing": "grayscale",
        "WebkitFontSmoothing": "antialiased",
        "display": "block",
        "fontFamily": "Lato, sans-serif",
        "fontSize": 20,
        "fontWeight": 700,
        "lineHeight": "24px",
        "marginBottom": 0,
        "marginTop": 0,
      }
    }
  >
    Subsequent requests:
  </h4>
  <div
    className=""
    style={
      Object {
        "alignItems": "stretch",
        "borderStyle": "solid",
        "borderWidth": 0,
        "boxSizing": "border-box",
        "display": "flex",
        "flexDirection": "column",
        "margin": 0,
        "minHeight": 0,
        "minWidth": 0,
        "padding": 0,
        "position": "relative",
        "zIndex": 0,
      }
    }
  >
    <span
      className=""
      style={
        Object {
          "MozOsxFontSmoothing": "grayscale",
          "WebkitFontSmoothing": "antialiased",
          "display": "block",
          "fontFamily": "Inconsolata, monospace",
          "fontSize": 17,
          "fontWeight": 400,
          "lineHeight": "22px",
        }
      }
    >
      get("an-id"): 
      uid--1-an-id
    </span>
    <span
      className=""
      style={
        Object {
          "MozOsxFontSmoothing": "grayscale",
          "WebkitFontSmoothing": "antialiased",
          "display": "block",
          "fontFamily": "Inconsolata, monospace",
          "fontSize": 17,
          "fontWeight": 400,
          "lineHeight": "22px",
        }
      }
    >
      get("something"): 
      uid--1-something
    </span>
  </div>
</div>
`;

exports[`wonder-blocks-core example 6 1`] = `
<div
  className=""
  style={
    Object {
      "alignItems": "stretch",
      "borderStyle": "solid",
      "borderWidth": 0,
      "boxSizing": "border-box",
      "display": "flex",
      "flexDirection": "column",
      "margin": 0,
      "minHeight": 0,
      "minWidth": 0,
      "padding": 0,
      "position": "relative",
      "zIndex": 0,
    }
  }
>
  <h4
    className=""
    style={
      Object {
        "MozOsxFontSmoothing": "grayscale",
        "WebkitFontSmoothing": "antialiased",
        "display": "block",
        "fontFamily": "Lato, sans-serif",
        "fontSize": 20,
        "fontWeight": 700,
        "lineHeight": "24px",
        "marginBottom": 0,
        "marginTop": 0,
      }
    }
  >
    First Provider with scope: first
  </h4>
  <div
    className=""
    style={
      Object {
        "alignItems": "stretch",
        "borderStyle": "solid",
        "borderWidth": 0,
        "boxSizing": "border-box",
        "display": "flex",
        "flexDirection": "column",
        "margin": 0,
        "minHeight": 0,
        "minWidth": 0,
        "padding": 0,
        "position": "relative",
        "zIndex": 0,
      }
    }
  >
    <span
      className=""
      style={
        Object {
          "MozOsxFontSmoothing": "grayscale",
          "WebkitFontSmoothing": "antialiased",
          "display": "block",
          "fontFamily": "Lato, sans-serif",
          "fontSize": 16,
          "fontWeight": 400,
          "lineHeight": "22px",
        }
      }
    >
      The id returned for "my-identifier": 
      uid-first-2-my-identifier
    </span>
  </div>
  <h4
    className=""
    style={
      Object {
        "MozOsxFontSmoothing": "grayscale",
        "WebkitFontSmoothing": "antialiased",
        "display": "block",
        "fontFamily": "Lato, sans-serif",
        "fontSize": 20,
        "fontWeight": 700,
        "lineHeight": "24px",
        "marginBottom": 0,
        "marginTop": 0,
      }
    }
  >
    Second Provider with scope: second
  </h4>
  <div
    className=""
    style={
      Object {
        "alignItems": "stretch",
        "borderStyle": "solid",
        "borderWidth": 0,
        "boxSizing": "border-box",
        "display": "flex",
        "flexDirection": "column",
        "margin": 0,
        "minHeight": 0,
        "minWidth": 0,
        "padding": 0,
        "position": "relative",
        "zIndex": 0,
      }
    }
  >
    <span
      className=""
      style={
        Object {
          "MozOsxFontSmoothing": "grayscale",
          "WebkitFontSmoothing": "antialiased",
          "display": "block",
          "fontFamily": "Lato, sans-serif",
          "fontSize": 16,
          "fontWeight": 400,
          "lineHeight": "22px",
        }
      }
    >
      The id returned for "my-identifier": 
      uid-second-3-my-identifier
    </span>
  </div>
</div>
`;

exports[`wonder-blocks-core example 7 1`] = `
<div
  className=""
  style={
    Object {
      "alignItems": "stretch",
      "borderStyle": "solid",
      "borderWidth": 0,
      "boxSizing": "border-box",
      "display": "flex",
      "flexDirection": "column",
      "margin": 0,
      "minHeight": 0,
      "minWidth": 0,
      "padding": 0,
      "position": "relative",
      "zIndex": 0,
    }
  }
>
  <span
    className=""
    style={
      Object {
        "MozOsxFontSmoothing": "grayscale",
        "WebkitFontSmoothing": "antialiased",
        "display": "block",
        "fontFamily": "Inconsolata, monospace",
        "fontSize": 17,
        "fontWeight": 400,
        "lineHeight": "22px",
      }
    }
  >
    interface IIdentifierFactory {
  </span>
  <div
    className=""
    style={
      Object {
        "alignItems": "stretch",
        "borderStyle": "solid",
        "borderWidth": 0,
        "boxSizing": "border-box",
        "display": "flex",
        "flexDirection": "row",
        "margin": 0,
        "minHeight": 0,
        "minWidth": 0,
        "padding": 0,
        "position": "relative",
        "zIndex": 0,
      }
    }
  >
    <div
      aria-hidden="true"
      className=""
      style={
        Object {
          "MsFlexBasis": "2em",
          "MsFlexPreferredSize": "2em",
          "WebkitFlexBasis": "2em",
          "alignItems": "stretch",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexBasis": "2em",
          "flexDirection": "column",
          "flexShrink": 0,
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "zIndex": 0,
        }
      }
    />
    <span
      className=""
      style={
        Object {
          "MozOsxFontSmoothing": "grayscale",
          "WebkitFontSmoothing": "antialiased",
          "display": "block",
          "fontFamily": "Inconsolata, monospace",
          "fontSize": 17,
          "fontWeight": 400,
          "lineHeight": "22px",
        }
      }
    >
      get(id: string): string;
    </span>
  </div>
  <span
    className=""
    style={
      Object {
        "MozOsxFontSmoothing": "grayscale",
        "WebkitFontSmoothing": "antialiased",
        "display": "block",
        "fontFamily": "Inconsolata, monospace",
        "fontSize": 17,
        "fontWeight": 400,
        "lineHeight": "22px",
      }
    }
  >
    }
  </span>
</div>
`;

exports[`wonder-blocks-core example 8 1`] = `
<div
  className=""
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": "plum",
      "borderStyle": "solid",
      "borderWidth": 0,
      "boxSizing": "border-box",
      "display": "flex",
      "flexDirection": "column",
      "margin": 0,
      "minHeight": 0,
      "minWidth": 0,
      "padding": 32,
      "position": "relative",
      "zIndex": 0,
    }
  }
>
  <span
    className=""
    style={
      Object {
        "MozOsxFontSmoothing": "grayscale",
        "WebkitFontSmoothing": "antialiased",
        "fontFamily": "sans-serif",
        "fontSize": 24,
      }
    }
  >
    Hello, world!
  </span>
  <div
    className=""
    style={
      Object {
        "alignItems": "stretch",
        "backgroundColor": "goldenrod",
        "borderStyle": "solid",
        "borderWidth": 0,
        "boxSizing": "border-box",
        "display": "flex",
        "flexDirection": "column",
        "margin": 0,
        "minHeight": 0,
        "minWidth": 0,
        "padding": 4,
        "position": "relative",
        "zIndex": 0,
      }
    }
  >
    Even with an array and inline styles!
  </div>
</div>
`;

exports[`wonder-blocks-core example 9 1`] = `
<div
  className=""
  style={
    Object {
      "alignItems": "stretch",
      "borderStyle": "solid",
      "borderWidth": 0,
      "boxSizing": "border-box",
      "display": "flex",
      "flexDirection": "column",
      "margin": 0,
      "minHeight": 0,
      "minWidth": 0,
      "padding": 0,
      "position": "relative",
      "zIndex": 0,
    }
  }
>
  <div
    className=""
    onClick={[Function]}
    style={
      Object {
        "alignItems": "stretch",
        "borderStyle": "solid",
        "borderWidth": 0,
        "boxSizing": "border-box",
        "display": "flex",
        "flexDirection": "column",
        "margin": 0,
        "minHeight": 0,
        "minWidth": 0,
        "padding": 0,
        "position": "relative",
        "zIndex": 0,
      }
    }
  >
    Click me!
  </div>
  <span
    aria-hidden="true"
    className=""
    style={
      Object {
        "MozOsxFontSmoothing": "grayscale",
        "WebkitFontSmoothing": "antialiased",
      }
    }
  >
    This text is hidden from screen readers.
  </span>
</div>
`;
